pkg/chunk/storage/factory.go

```diff
@@ -180,7 +180,7 @@ func NewStore(cfg Config, storeCfg chunk.StoreConfig, schemaCfg chunk.SchemaConf
 }
 
 // NewIndexClient makes a new index client of the desired type.
-func NewIndexClient(name string, cfg Config, schemaCfg chunk.SchemaConfig) (chunk.IndexClient, error) {
+func NewIndexClient(name string, cfg Config, schemaCfg chunk.SchemaConfig, purpose string, registerer prometheus.Registerer) (chunk.IndexClient, error) {
```

Comment:
Lets not pass a string here but the DayTime.  And lets not pass the registerer here, but wrap a new registerer next to the instantiation of the Cassandra client.  Otherwise I feel it sets expectation that the registerer will be used for all client.s

Reply:
The problem is `NewIndexClient` is also used to create `IndexClient` for storing delete requests [here](https://github.com/cortexproject/cortex/blob/92622a86800c5f5d12eda1cc74db030413f866d3/pkg/cortex/modules.go#L336), which is not tied to start date.
I added the `purpose` label to track the sessions based on the purpose of usage which could be for storing series index(using Daytime for this since there are multiple schemas) or for indexing delete requests or any other use case that we may have in future.

Reply:
In that case leave the registerer plumbed through but don't plumb reason though - just wrap the registerer before each instantiation of NewIndexClient.

Reply:
We can create the registerer with the `purpose` or something label for all the clients if you think that is more appropriate and clearer. I think there is no harm in having an extra-label.

Reply:
I don't mind having the extra label (its needed) but there is no point in plumbing it through like that.  We should add it close to where it semantically makes sense.

Reply:
@tomwilkie I have done these changes and also made changes to use the same registerer for metrics exported by dynamodb. Thanks for the review!